src/components/TagInput/TagInput.tsx

```diff
@@ -0,0 +1,69 @@
+import { FormEvent, KeyboardEvent, useState } from 'react';
+
+import { TextInput } from '../Input';
+import { Tag, TagGroup } from '../Tag';
+import { Merge } from '../../types/polymophic';
+
+export type TagInputBaseProps = {
+  hint?: string;
+  label: string;
+  onTagsChange(tags: Array<string>): any;
+  tags?: Array<string>;
+};
+
+export type TagInputProps = Merge<React.InputHTMLAttributes<HTMLInputElement>, TagInputBaseProps>;
+
+export const TagInput = ({
+  hint,
+  label,
+  onTagsChange,
+  tags,
+  ...props
+}: TagInputProps) => {
+  const [input, setInput] = useState('');
+  const [values, setValues] = useState(tags);
+
+  const handleKeyDown = (e: KeyboardEvent) => {
+    if (e.key === "Enter" && input.trim()) {
+      e.preventDefault();
+      if (values?.includes(input.trim())) return;
+      const newValues = [...(values ?? []), input.trim()];
+      setValues(newValues);
+      setInput('');
+      onTagsChange(newValues);
+    }
+  };
+
+  const handleDeleteClick = (tag: string) => {
+    const newValues = [...(values ?? []).filter((el) => el !== tag)];
+    setValues(newValues);
+    onTagsChange(newValues);
+  };
+
+  return (
+    <div>
+      <TextInput
+        hint={hint}
+        label={label}
+        onChange={(event: FormEvent<HTMLInputElement>) => setInput((event?.target as HTMLTextAreaElement)?.value)}
+        onKeyDown={(event: KeyboardEvent<HTMLInputElement>) => handleKeyDown(event)}
+        type="text"
+        value={input}
+        {...props}
+      />
+      <TagGroup>
+        {(values ?? []).map((tag) => (
+          <Tag
+            className="fr-mr-1w"
+            key={tag}
+            onClick={() => handleDeleteClick(tag)}
+            title={`Mot (${tag}) sélectionné, cliquer sur la touche "Entrée" ou la touche "Espace" pour le supprimer.`}
+          >
+            {tag}
+            <span className='fr-icon-close-line' />
```

Comment:
better not to use a span here 
best alternative : add css ```.custom-dismissible-tag {
  &::after {
    flex: 0 0 auto;
    display: inline-block;
    vertical-align: calc((.75em - var(--icon-size)) * .5);
    background-color: currentColor;
    width: var(--icon-size);
    height: var(--icon-size);
    -webkit-mask-size: 100% 100%;
    mask-size: 100% 100%;
    -webkit-mask-image: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0IiBoZWlnaHQ9IjI0Ij48cGF0aCBkPSJtMTIgMTAuNiA0Ljk1LTQuOTYgMS40IDEuNEwxMy40MiAxMmw0Ljk2IDQuOTUtMS40IDEuNEwxMiAxMy40MmwtNC45NSA0Ljk2LTEuNC0xLjRMMTAuNTggMTIgNS42MyA3LjA1bDEuNC0xLjR6Ii8+PC9zdmc+);
    mask-image: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0IiBoZWlnaHQ9IjI0Ij48cGF0aCBkPSJtMTIgMTAuNiA0Ljk1LTQuOTYgMS40IDEuNEwxMy40MiAxMmw0Ljk2IDQuOTUtMS40IDEuNEwxMiAxMy40MmwtNC45NSA0Ljk2LTEuNC0xLjRMMTAuNTggMTIgNS42MyA3LjA1bDEuNC0xLjR6Ii8+PC9zdmc+);
    --icon-size: 1rem;
    content: "";
    margin-left: .25rem;
    margin-right: -.25rem;
  }
}```  in DismissableTag and use DismissableTag here. It is a button by default
Ok alternative : use the css directly in InputTag and set as="button"
Not so good alternative : add fr-icon--sm and aria-hidden